lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -188,26 +233,32 @@ sub beacon_query {
   }
  
   # check variant if only all parameters are valid
-  if(!defined($beaconError)){ 
+  if(!defined($beaconError)){
     # Check allele exists 
     my $reference_name = $data->{referenceName};
-    my $start = $data->{start};
-    my $end = $data->{end} ? $data->{end} : $data->{start}; 
     my $ref_allele = $data->{referenceBases};
     my $alt_allele = $data->{alternateBases} ? $data->{alternateBases} : $data->{variantType};
 
-    # Currently assumes only 1 dataset
-    # TODO Multiple dataset - for each dataset
+    # SNV or small indels have a start (or start-end)
+    # Structural variants can have start-end or startMin/startMax-endMin/endMax
+    my $start = $data->{start} ? $data->{start} : $data->{startMin};
+    my $start_max = $data->{startMax} ? $data->{startMax} : $start;
+    my $end;
+    if($data->{end}){ $end = $data->{end}; }
+    elsif($data->{endMax}){ $end = $data->{endMax}; }
+    else{ $end = $start; }
+    my $end_min = $data->{endMin} ? $data->{endMin} : $end;
+
+    # Multiple dataset
     # check variant exists and report exists overall
-    #
-    my $dataset = $self->get_beacon_dataset($db_meta);
-
     my ($exists, $dataset_response)  = $self->variant_exists($reference_name,
                                        $start,
+                                       $start_max,
                                        $end,
+                                       $end_min,
                                        $ref_allele,
                                        $alt_allele,
-                                       $incl_ds_response, $dataset);
+                                       $incl_ds_response, $assemblyId, \@dataset_ids_list, $has_dataset);
```

Comment:
number of parameters is growing a bit too much...
is it worth refactoring to a param hash?
it can at least be formatted properly (either all in sequence, or all one per line)